crates/oxc_regexp_parser/src/parser/literal_parser.rs

```diff
@@ -0,0 +1,119 @@
+use oxc_allocator::Allocator;
+use oxc_diagnostics::{OxcDiagnostic, Result};
+use oxc_syntax::identifier::is_line_terminator;
+
+use crate::{
+    ast,
+    ast_builder::AstBuilder,
+    parser::{
+        body_parser::PatternParser, flag_parser::FlagsParser, options::ParserOptions,
+        reader::Reader, span::SpanFactory,
+    },
+};
+
+// LiteralParser
+pub struct Parser<'a> {
+    allocator: &'a Allocator,
+    source_text: &'a str,
+    options: ParserOptions,
+    ast: AstBuilder<'a>,
+    span_factory: SpanFactory,
+}
+
+impl<'a> Parser<'a> {
+    pub fn new(allocator: &'a Allocator, source_text: &'a str, options: ParserOptions) -> Self {
+        Self {
+            allocator,
+            source_text,
+            options,
+            ast: AstBuilder::new(allocator),
+            span_factory: SpanFactory::new(options.span_offset),
+        }
+    }
+
+    // NOTE: Should return `ParserReturn { (empty)literal, errors }`?
+    pub fn parse(self) -> Result<ast::RegExpLiteral<'a>> {
```

Comment:
Are we building a recoverable parser? Can `Result` if this parser is not recoverable.

Reply:
Probably no? (I don't think I even properly understood what recoverable meant in the first place, being inexperienced...)

This comment was a remnant of the process of following the code of `oxc_parser`.